src/open_r1/rewards.py

```diff
@@ -62,12 +62,34 @@ def accuracy_reward(completions, solution, **kwargs):
 
 def format_reward(completions, **kwargs):
     """Reward function that checks if the reasoning process is enclosed within <think> and </think> tags, while the final answer is enclosed within <answer> and </answer> tags."""
-    pattern = r"^<think>.*?</think>\s*<answer>.*?</answer>$"
+    pattern = r"^<think>\n.*?\n</think>\n<answer>\n.*?\n</answer>$"
```

Comment:
Note that this is also aligned with the system prompt we use for the instruct models